### PR TITLE
enhance(datastore): support server side flush

### PR DESCRIPTION
**Pull request by xuchuan**
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java [hunk 1]
```diff
@@ -385,6 +420,13 @@ private PopulationStatus populateOutput() {
                 }
                 entry = this.entriesToWrite.getFirst();
             }
+            if (entry.getId() == FLUSH_ID) {
+                synchronized (this.entriesToWrite) {
+                    this.entriesToWrite.removeFirst();
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:425]: It may delete the unexpected entry?

Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:425]: FLUSH_ID = -12345, which will never appear in normal entries.

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:425]: I didn't make myself clear, I mean that `getFirst` and `removeFirst` are not in the same synchronized block.

Comment (xuchuan) [reply server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:425]: entriesToWrite is a queue. Entries are removed from the front and appended to the back. Only populateOutput may remove entries, so it is not a problem.

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**